modules/gateway/upnp.go

```diff
@@ -63,6 +63,9 @@ func (g *Gateway) threadedLearnHostname() {
 	if err != nil {
 		host, err = myExternalIP()
```

Comment:
we should have this check be last I think, and also have it only run if !build.Debug, that way debug builds of Sia get to see what happens if the centralized version doesn't exist as an alternative.

Reply:
I put the peer-to-peer discovery last because I think that we should give the network some time to upgrade to the new version. Otherwise the users who upgrade first will always experience slow ip discovery if they don't have a UPnP enabled router until the majority of the network supports the new RPC.
The next step would be to have `myExternalIP` as a fallback in a future update and the last one is to completely remove it once 99.99% of the network support the RPC.